core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceFactory.java

```diff
@@ -750,18 +759,21 @@ static String findDescription(AnnotatedElement annotatedElement) {
             final Class<?> clazz = executable.getDeclaringClass();
             final String fileName = getFileName(clazz.getCanonicalName());
             final String propertyName = executable.getName() + '.' + parameter.getName();
+            final Properties cachedProperties = DOCUMENTATION_PROPERTIES_CACHE.getIfPresent(fileName);
+            if (cachedProperties != null) {
+                return cachedProperties.getProperty(propertyName);
+            }
             try (InputStream stream = AnnotatedServiceFactory.class.getClassLoader()
                                                                    .getResourceAsStream(fileName)) {
                 if (stream == null) {
                     return null;
                 }
                 final Properties properties = new Properties();
                 properties.load(stream);
-                if (properties.containsKey(propertyName)) {
-                    return properties.getProperty(propertyName);
-                }
+                DOCUMENTATION_PROPERTIES_CACHE.put(fileName, properties);
+                return properties.getProperty(propertyName);
             } catch (IOException exception) {
-                exception.printStackTrace();
+                logger.warn("Error loading API description file from FS. File: " + fileName, exception);
```

Comment:
- ` from FS. File` could be removed for brevity:
- Missing `an`
- Need to use `{}` instead of concatenation

      "Error loading an API description file: {}", fileName, exception

Reply:
I have to use the concatenation because otherwise the logger thinks the exception is a second argument for the string...

Reply:
Really? I don't think so. This should work as expected for sure:

    logger.warn("Failed to load an API description file: {}", fileName, exception);

Reply:
Unlike the Javadoc and the method signature, Logging frameworks differently works if the last parameter is instance of `Throwable`.

https://github.com/qos-ch/logback/blob/7ee74ad132bf74ead68c5bb553f5353aa2588bb0/logback-classic/src/main/java/ch/qos/logback/classic/Logger.java#L701-L703

https://github.com/qos-ch/logback/blob/7ee74ad132bf74ead68c5bb553f5353aa2588bb0/logback-classic/src/main/java/ch/qos/logback/classic/spi/LoggingEvent.java#L126-L128